schema/jsonschema/x-tyk-gateway.json

```diff
@@ -0,0 +1,1159 @@
+      "type": "object",
+      "required": [
+        "domain",
+        "publicKeys"
+      ]
+    },
+    "PinnedPublicKeys": {
+      "items": {
+        "$ref": "#/$defs/PinnedPublicKey"
+      },
+      "type": "array"
+    },
+    "PluginBundle": {
+      "properties": {
+        "enabled": {
+          "type": "boolean"
+        },
+        "path": {
+          "type": "string"
+        }
+      },
+      "additionalProperties": false,
+      "type": "object",
+      "required": [
+        "enabled",
+        "path"
+      ]
+    },
+    "PluginConfig": {
+      "properties": {
+        "driver": {
+          "type": "string"
+        },
+        "bundle": {
+          "$ref": "#/$defs/PluginBundle"
+        },
+        "data": {
+          "$ref": "#/$defs/PluginConfigData"
+        }
+      },
+      "additionalProperties": false,
+      "type": "object"
+    },
+    "PluginConfigData": {
+      "properties": {
+        "enabled": {
+          "type": "boolean"
+        },
+        "value": {
+          "type": "object"
+        }
+      },
+      "additionalProperties": false,
+      "type": "object",
+      "required": [
+        "enabled",
+        "value"
+      ]
+    },
+    "PostAuthenticationPlugin": {
+      "properties": {
+        "plugins": {
+          "$ref": "#/$defs/CustomPlugins"
+        }
+      },
+      "additionalProperties": false,
+      "type": "object"
+    },
+    "PostPlugin": {
+      "properties": {
+        "plugins": {
+          "$ref": "#/$defs/CustomPlugins"
+        }
+      },
+      "additionalProperties": false,
+      "type": "object"
+    },
+    "PrePlugin": {
+      "properties": {
+        "plugins": {
+          "$ref": "#/$defs/CustomPlugins"
+        }
+      },
+      "additionalProperties": false,
+      "type": "object"
+    },
+    "Provider": {
+      "properties": {
+        "issuer": {
+          "type": "string"
+        },
+        "clientToPolicyMapping": {
+          "items": {
+            "$ref": "#/$defs/ClientToPolicy"
+          },
+          "type": "array"
+        }
+      },
+      "additionalProperties": false,
+      "type": "object"
+    },
+    "ResponsePlugin": {
+      "properties": {
+        "plugins": {
+          "$ref": "#/$defs/CustomPlugins"
+        }
+      },
+      "additionalProperties": false,
+      "type": "object"
+    },
+    "ScopeToPolicy": {
+      "properties": {
+        "scope": {
+          "type": "string"
+        },
+        "policyId": {
+          "type": "string"
+        }
+      },
+      "additionalProperties": false,
+      "type": "object"
+    },
+    "Scopes": {
+      "properties": {
+        "claimName": {
+          "type": "string"
+        },
+        "scopeToPolicyMapping": {
+          "items": {
+            "$ref": "#/$defs/ScopeToPolicy"
+          },
+          "type": "array"
+        }
+      },
+      "additionalProperties": false,
+      "type": "object"
+    },
+    "SecuritySchemes": {
```

Comment:
this is a limitation we currently have since SecuritySchemes is 
`type SecuritySchemes map[string]interface{}`, can we use [custom type definitions](https://github.com/invopop/jsonschema#custom-type-definitions) to have `anyOf` like this https://github.com/TykTechnologies/tyk/blob/master/apidef/oas/schema/x-tyk-api-gateway.json#L897?

Reply:
We can, but we'd need to apply them from something like x-tyk-gateway.overrides.json ; can't figure out the types, because SecuritySchemes is unsafe, and we can't (yet?) know which types should fill it as the interface is basically `any`.